k8s_pod_filter_test.go

```diff
@@ -142,5 +142,154 @@ func Test_ShouldTailLogs(t *testing.T) {
+			filter := NewTailAllFilter("beowulf.example.com", 443, 10*time.Millisecond, "/tmp/does-not-exist")
+
+			So(filter, ShouldBeNil)
+		})
+	})
+}
+
+func Test_TailAllFilter_ShouldTailLogs(t *testing.T) {
+	Convey("TailAllFilter.ShouldTailLogs()", t, func() {
+		Reset(func() { httpmock.DeactivateAndReset() })
+
+		filter := NewTailAllFilter("beowulf.example.com", 80, 10*time.Millisecond, credsPath)
+		httpmock.ActivateNonDefault(filter.client)
+
+		Convey("returns true by default when no annotation is set", func() {
+			httpmock.RegisterResponder("GET", "=~http://beowulf.example.com:80/api/v1/namespaces/the-awesome-place/pods.*",
+				func(req *http.Request) (*http.Response, error) {
+					return httpmock.NewStringResponse(200, `{"items":[{"metadata":{"annotations":{}}}]}`), nil
+				},
+			)
+
+			pod := &Pod{
+				Name:        "awesome-pod",
+				ServiceName: "awesome-pod",
+				Namespace:   "the-awesome-place",
+			}
+
+			shouldTail, err := filter.ShouldTailLogs(pod)
+			So(err, ShouldBeNil)
+			So(shouldTail, ShouldBeTrue)
+		})
+
+		Convey("respects community.com/TailLogs=false annotation", func() {
+			httpmock.RegisterResponder("GET", "=~http://beowulf.example.com:80/api/v1/namespaces/the-awesome-place/pods.*",
+				func(req *http.Request) (*http.Response, error) {
+					return httpmock.NewStringResponse(200, `{"items":[{"metadata":{"annotations":{"community.com/TailLogs":"false"}}}]}`), nil
+				},
+			)
+
+			pod := &Pod{
+				Name:        "awesome-pod",
+				ServiceName: "awesome-pod",
+				Namespace:   "the-awesome-place",
+			}
+
+			shouldTail, err := filter.ShouldTailLogs(pod)
+			So(err, ShouldBeNil)
+			So(shouldTail, ShouldBeFalse)
+		})
+
+		Convey("ignores community.com/TailLogs=true annotation (still returns true)", func() {
+			httpmock.RegisterResponder("GET", "=~http://beowulf.example.com:80/api/v1/namespaces/the-awesome-place/pods.*",
+				func(req *http.Request) (*http.Response, error) {
+					return httpmock.NewStringResponse(200, `{"items":[{"metadata":{"annotations":{"community.com/TailLogs":"true"}}}]}`), nil
+				},
+			)
+
+			pod := &Pod{
+				Name:        "awesome-pod",
+				ServiceName: "awesome-pod",
+				Namespace:   "the-awesome-place",
+			}
+
+			shouldTail, err := filter.ShouldTailLogs(pod)
+			So(err, ShouldBeNil)
+			So(shouldTail, ShouldBeTrue)
+		})
+
+		Convey("returns true when no pods found (unknown pod)", func() {
+			httpmock.RegisterResponder("GET", "=~http://beowulf.example.com:80/api/v1/namespaces/the-awesome-place/pods.*",
+				func(req *http.Request) (*http.Response, error) {
+					return httpmock.NewStringResponse(200, `{"items":[]}`), nil
+				},
+			)
+
+			pod := &Pod{
+				Name:        "unknown-pod",
+				ServiceName: "unknown-pod",
+				Namespace:   "the-awesome-place",
+			}
+
+			shouldTail, err := filter.ShouldTailLogs(pod)
+			So(err, ShouldBeNil)
+			So(shouldTail, ShouldBeTrue)
+		})
+	})
+}
+
+func Test_DiscoveryFilter_Interface(t *testing.T) {
```

Comment:
Ah crap, I see there is already an interface. Just use that, then instead of `PodFilter`.